File: JobHunter/ClientApp/src/app/models/jobOffer.ts
import { Users } from "./users";
import { BidOffer } from "./BidOffer";

export class JobOffer {
  public  id:number;
public title:string
public  declaredCost:number;
public  description :string;
public  status :number;
public takenById :number;
public   takenBy:Users;
public addedById :number;
public  addedBy:Users;
public endOfferDate :Date;
  public categoryId: number;
  public category: Category;

public bidOffers:BidOffer[];
  public endedAs: number;
  public edited: boolean;

  public bidding:boolean
}
export  class TakenOffer {
  public id: number;
  public title: string
  public declaredCost: number;
  public description: string;
  public status: number;
  public takenById: number;
  public takenBy: Users;
  public addedById: number;
  public addedBy: Users;
  public finishDate: Date;
}
export class Category {
  public id: number;
  public description: string;
}
export class EndModel {
  public offerId: number;
  public statusId: number;
}
/*
 JobOffer
 1- do wzięcia
 2- wzięte
3- zakończone z rekomendacją
4- zakończone
5 - zakończone z "reklamacją"
*/

File: Services/VM/JobOffer.cs
using Model;
using System;
using System.Collections.Generic;
using System.Text;

namespace Services.VM
{
    public class JobOffer
    {

        public int Id { get; set; }

        public string Title { get; set; }
        public decimal DeclaredCost { get; set; }
        public string Description { get; set; }
        public int Status { get; set; }

      public int? TakenById { get; set; }

        public virtual Users TakenBy { get; set; }

        public int AddedById { get; set; }

        public virtual Users AddedBy { get; set; }
        public DateTime EndOfferDate { get; set; }


        public ICollection<BidOffer> BidOffers { get; set; }
        public int? EndedAs { get; set; }
   public int CategoryId { get; set; }

        public virtual Category Category { get; set; }
    }
}
